ksql-engine/src/test/java/io/confluent/ksql/util/KafkaTopicClientTest.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.DeleteTopicsResult;
+import org.apache.kafka.clients.admin.DescribeClusterResult;
+import org.apache.kafka.clients.admin.DescribeConfigsResult;
+import org.apache.kafka.clients.admin.DescribeTopicsResult;
+import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartitionInfo;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.kafka.common.internals.KafkaFutureImpl;
+import org.easymock.Mock;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import io.confluent.ksql.exception.KafkaTopicException;
+
+import static org.easymock.EasyMock.*;
+
+public class KafkaTopicClientTest {
+
+  private AdminClient adminClient;
+
+  private KafkaTopicClient kafkaTopicClient;
+
+  String topicName1 = "topic1";
+  String topicName2 = "ksql_query_2-KSTREAM-MAP-0000000012-repartition";
+  String topicName3 = "ksql_query_2-KSTREAM-REDUCE-STATE-STORE-0000000003-changelog";
+
+  @Before
+  public void init() throws ExecutionException, InterruptedException, IllegalAccessException,
```

Comment:
This needs to be broken out into methods that are understandable. It is really difficult to tell what is going on here. Also this shouldn't all be done in the before method. Not all of the expectations apply to every test.
I think line 69 - line 79 is to do with the describe cluster stuff in init. So make a method for it.

Reply:
@dguy Refactored the whole test to apply your feedback.